lms/djangoapps/instructor/static/instructor/ReportPopup/components/Popup/PopupModal.jsx

```diff
@@ -0,0 +1,78 @@
+import * as React from 'react';
+import * as PropTypes from 'prop-types';
+import {Modal, Button, Hyperlink} from '@edx/paragon/static';
+
+export default class PopupModal extends React.Component {
+    constructor(props) {
+        super(props);
+        this.handleDownload = this.handleDownload.bind(this);
+    }
+
+    handleDownload() {
+        window.open(`/media/${this.props.reportPath}`, '_blank');
+    }
+
+    render() {
```

Comment:
You can break this method down into ``getButtons``, ``getBody``, ``getProgress`` etc.